server/routes/bookVisit/checkVisitDetailsController.ts

```diff
@@ -47,12 +48,20 @@ export default class CheckVisitDetailsController {
         delete req.session.bookingJourney
         return res.redirect(paths.BOOK_VISIT.BOOKED)
       } catch (error) {
-        // HTTP 400 Bad Request is the response when a session is no longer available
-        if (error.status === 400) {
-          req.flash('message', 'Your visit time is no longer available. Select a new time.')
-          delete bookingJourney.selectedVisitSession
-          return res.redirect(paths.BOOK_VISIT.CHOOSE_TIME)
+        if (error.status === 422 && error.validationErrors?.length) {
```

Comment:
It's likely that these checks and new error code responses will be added to the create/change application in the future. When they are, these conditions can be moved to a separate helper function to avoid duplication.